resources/js/package.js

```diff
@@ -1,37 +1,69 @@
-document.addEventListener('turbo:load', function () {
-    ['shipping_address', 'billing_address'].forEach((type) => {
-        window.app.$watch('checkout.'+type+'.postcode', function () {
-            getAddressFromPostcodeservice(type)
-        })
+import { useThrottleFn } from "@vueuse/core"
+
+const postcodeObserver = new MutationObserver(useThrottleFn(() => {
+    let types = ['housenumber', 'postcode', 'country']
+    types.forEach((type) => {
+        document.querySelectorAll(`[id\$='_${type}']`).forEach(el => {
+            if (el.id in window.app.custom.postcode_data) {
+                return
+            }
 
-        window.app.$watch('checkout.'+type+'.street.1', function () {
-            getAddressFromPostcodeservice(type)
+            window.app.custom.postcode_data[el.id] = el.value
+            let id = el.id.replace(`_${type}`, '')
+            el.addEventListener('change', () => {
+                window.app.custom.postcode_data[el.id] = el.value
+                getAddressFromPostcodeservice(id)
+            })
         })
     })
+}, 500, true, true))
+
+document.addEventListener('turbo:load', function () {
+    window.app.custom.postcode_data = {}
+
+    postcodeObserver.observe(window.app.$el, {
+        childList: true,
+        subtree: true,
+    })
 })
 
-async function getAddressFromPostcodeservice(type) {
-    if (window.app.checkout[type].country_id != 'NL') {
+async function getAddressFromPostcodeservice(id) {
+    let get = (type) => window.app.custom.postcode_data[`${id}_${type}`]
+    let elm = (type) => document.getElementById(`${id}_${type}`)
+    let set = (type, value) => {
+        let el = elm(type)
+        if (el) {
+            el.value = value
+            el.dispatchEvent(new Event('change'))
```

Comment:
This might create an infinite loop with the eventlistener above

Reply:
This is true, in this case I only use this set() function on the fields that don't get watched by the eventlistener above but it might be a good idea to add some safety either way.